source/code/include/scxsystemlib/scxsysteminfo.h

```diff
@@ -133,6 +133,7 @@ namespace SCXSystemLib
         std::wstring GetDefaultSudoPath() const;
         std::wstring GetShellCommand(const std::wstring& command) const;
         std::wstring GetElevatedCommand(const std::wstring& command) const;
+	    std::string GetDefaultTmpDir() const;
```

Comment:
I don't think you should implement this. You should have SetDefaultTemporaryDirectory() to set the default for the project using the pal, and then GetTemporaryDirectory (which needs to return a wstring) to fetch it.

Also, this is new functionality. You need to have unit tests for them. Please add unit tests, and I'll need to see another iteration including unit tests.